### PR TITLE
[MM-24143] Adding circleci configuration to automatically zip and push lambda functions and move lambda functions under a single directory

### PR DESCRIPTION
**Pull request by stylianosrigas**
MM-24143 

- Adding CircleCI configuration to automatically zip and push lambda functions
- Move lambda functions under a single directory

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,274 @@
+version: 2.1
+
+orbs:
+  aws-s3: circleci/aws-s3@1.0.11
+
+jobs:
+  grafana-aws-metrics-master:
+    docker:
```
Comment (cpanato) [inline .circleci/config.yml:8]: remove this part that is the same for all jobs and do this

in the top of the file
```
executors:
  default:
    docker:
    - image: circleci/golang:1.14.1
    working_directory: /go/src/github.com/mattermost/mattermost-cloud-monitoring
```

for each job that uses that image
```
grafana-aws-metrics-master:
   executor:
       name: default
...
```

**Comment by cpanato** [comment]
Also we might want to remove the `.zip` files from the repo as well

File: .circleci/config.yml (hunk 1 above)
Comment (stylianosrigas) [reply .circleci/config.yml:8]: Done, together with some wider refactoring.

**Comment by stylianosrigas** [comment]
@carlos Refactoring done and zip files removed.

**Pull request merged**